node/full_node_integration_test.go

```diff
@@ -23,13 +26,86 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/rollkit/rollkit/config"
+	"github.com/rollkit/rollkit/da"
 	test "github.com/rollkit/rollkit/test/log"
 	"github.com/rollkit/rollkit/test/mocks"
 	"github.com/rollkit/rollkit/types"
 
 	testutils "github.com/celestiaorg/utils/test"
 )
 
+func TestCentralizedSequencer(t *testing.T) {
+	require := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	genDoc, privkey := types.GetGenesisWithPrivkey()
+	genDoc.AppHash = make([]byte, 32)
+	nodeKey := &p2p.NodeKey{
+		PrivKey: privkey,
+	}
+	signingKey, err := types.GetNodeKey(nodeKey)
+	require.NoError(err)
+
+	app := &mocks.Application{}
+	app.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
+	app.On("CheckTx", mock.Anything).Return(abci.ResponseCheckTx{})
+	app.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	app.On("DeliverTx", mock.Anything).Return(abci.ResponseDeliverTx{})
+	app.On("EndBlock", mock.Anything).Return(abci.ResponseEndBlock{})
+	app.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+
+	vals := types.GetValidatorSetFromGenesis(genDoc)
+
+	dalc := getMockDA()
+
+	blockManagerConfig := config.BlockManagerConfig{
+		BlockTime:     1 * time.Second,
+		DAStartHeight: 1,
+		DABlockTime:   1 * time.Second,
+	}
+	node, err := newFullNode(ctx, config.NodeConfig{DAAddress: MockServerAddr, Aggregator: false, BlockManagerConfig: blockManagerConfig}, signingKey, signingKey, proxy.NewLocalClientCreator(app), genDoc, log.TestingLogger())
+	require.NoError(err)
+	node.dalc = dalc
+	node.blockManager.SetDALC(dalc)
+
+	err = node.Start()
+	require.NoError(err)
+
+	lastState, err := node.Store.GetState()
+	require.NoError(err)
+	lastResults := lastState.LastResultsHash
+
+	validBlock, err := types.GetFirstBlock(privkey, &vals, lastResults, false, false)
+	require.NoError(err)
+	err = validBlock.ValidateBasic()
+	require.NoError(err)
+	junkProposerBlock, err := types.GetFirstBlock(privkey, &vals, lastResults, true, false)
+	require.NoError(err)
+	err = junkProposerBlock.ValidateBasic()
+	require.Error(err)
+	sigInvalidBlock, err := types.GetFirstBlock(privkey, &vals, lastResults, false, true)
+	require.NoError(err)
+	err = sigInvalidBlock.ValidateBasic()
+	require.Error(err)
+	submitResp := dalc.SubmitBlocks(ctx, []*types.Block{validBlock, junkProposerBlock, sigInvalidBlock})
+	fmt.Println(submitResp)
+	require.Equal(submitResp.Code, da.StatusSuccess)
+
+	require.NoError(testutils.Retry(3000, 1*time.Millisecond, func() error {
```

Comment:
In my experience it is better to have a longer interval time in the retry to give the machine more time to change state. 

```suggestion
	require.NoError(testutils.Retry(30, 100*time.Millisecond, func() error {
```